fe/fe-core/src/main/java/com/starrocks/sql/optimizer/task/OptimizerTask.java

```diff
@@ -59,6 +59,9 @@ void filterInValidRules(GroupExpression groupExpression,
             if (optimizerConfig.isRuleDisable(rule.type())) {
                 continue;
             }
+            if (rule.exhausted(context.getOptimizerContext())) {
```

Comment:
why not check it in Rule.check? then if any rule which want to consider exhausted should call super.check.
So If we add more RuleTask can naturally use this property without any more modification.